### ISSUE TITLE
Display number of users in a Girder instance

### ISSUE THREAD

**Issue by mgrauer** [girder/girder#2251]
I have a need to display the number of users in a girder instance, to admin users only, in the web client.  This could be just a number in parenthesis next to the users nav link, e.g.  

    Users (42)

I'd be fine with a generalization of this, e.g. maybe it could be enabled/disabled by a config setting, or available to all or only admin users as a config setting.

**Comment by brianhelba** [comment]
For site admin users, who can see all users, this is actually a very cheap and easy operation to perform: we just count the number of documents in the `user` collection.

If we want to support non-admin users seeing this, we'd have to do standard access filtering, since users are access controlled and could be private in theory. To do this on every user in the DB is likely expensive enough that we should probably at least do it in an aggregation pipeline, if we allow it at all.

The only open question is where we should expose this info. To be consistent with the other endpoints that list `nFolders`, `nItems`, etc., I'd propose that we make a `/user/details` endpoint that will return an object like:
```
{
    'nUsers': 42
}
```

@girder/developers Any thoughts about this?

**Comment by jeffbaumes** [comment]
I've always thought it a bit odd that normal users can see the full user list (that's the current default behavior, right?). I don't know why it is needed, since users can be discovered via the search menu. It seems like the user list is mostly for admins, and even seems like a slight security issue being able to enumerate all users.

So, I'd suggest only admins to see/access the Users sidebar item, and the (full) number of users can be displayed in parentheses.

**Comment by brianhelba** [comment]
@jeffbaumes That would be a major change, which I'd also have some thoughts about. Can you file that (`Hide user list from normal users`) as a separate issue, so we can cleanly debate the scope and implications there?

**Comment by jeffbaumes** [comment]
To scope down my thoughts, I think @brianhelba's `nUsers` field should only be admin-accessible. Why would a normal user want to know how many users they have access to see? And, as @brianhelba notes, this case would be more complex to implement.

**Comment by brianhelba** [comment]
Until #2252 goes somewhere, I see nothing wrong with exposing the count functionality to all users if we can do so with reasonable performance. The alternative (which you can do right now), is just to page through the users list, manually counting the number of users until you reach the end.

**Comment by Pierre-Assemat** [comment]
New pull request in working progress #2262

**Comment by Pierre-Assemat** [comment]
Once the server side is done, what are the expectation on girder client side ?

**Comment by brianhelba** [comment]
I think putting something at the top of the `#users` page (`UsersView`) would be the simplest and best.

The alternative would be to embed the count as a badge inside the global nav container (the pane with Collections, Users, etc. on the right side), However, I don't trust that making a REST request every time we render that container is a good idea:
* I'm particularly worried about how Girder is overly aggressive about calling `cancelRestRequests` in top-level views)
* If we ever make this functionality usable for non-admin users, it's going to result in a lot more server requests just to load the app
* It's not clear how often this should be refreshed, or how a user would manually trigger a refresh (without reloading the whole web app)
* It would be inconsistent with the fact that we don't display a badge for Collections and Groups (yet)

**Comment by danlamanna** [comment]
> It's not clear how often this should be refreshed, or how a user would
> manually trigger a refresh (without reloading the whole web app)

I don't think we should do this, but if we were I think this would make
sense being built in to the notification stream. This is how Girder deals
with new jobs/status updates in the job list pane.

**Comment by mgrauer** [comment]
It satisfies my needs as long as an admin user can easily navigate to it, so on the Users page is fine.

**Comment by brianhelba** [comment]
@danlamanna I agree, and I think we'd need pub-sub events (probably mediated over websockets) for us to start making these sort of "global site" events down to every single user.

**Comment by danlamanna** [comment]
Related #548.

**Comment by mgrauer** [comment]
Also, for my purposes, updating this value is out of scope, displaying it once when the page is rendered is sufficient.

**Comment by brianhelba** [comment]
Fixed by #2262
